src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.OpenFileTracker.cs

```diff
@@ -83,13 +91,35 @@ public sealed class OpenFileTracker : IRunningDocumentTableEventListener
             /// This cutoff of 10 was chosen arbitrarily and with no evidence whatsoever.</remarks>
             private const int CutoffForCheckingAllRunningDocumentTableDocuments = 10;
 
-            private OpenFileTracker(VisualStudioWorkspaceImpl workspace, IVsRunningDocumentTable runningDocumentTable, IComponentModel componentModel)
+            private OpenFileTracker(
+                VisualStudioWorkspaceImpl workspace,
+                IThreadingContext threadingContext,
+                IVsRunningDocumentTable runningDocumentTable,
+                IComponentModel componentModel,
+                IAsynchronousOperationListenerProvider listenerProvider)
             {
                 _workspace = workspace;
+                _threadingContext = threadingContext;
                 _foregroundAffinitization = new ForegroundThreadAffinitizedObject(workspace._threadingContext, assertIsForeground: true);
                 _asyncOperationListener = componentModel.GetService<IAsynchronousOperationListenerProvider>().GetListener(FeatureAttribute.Workspace);
                 _runningDocumentTableEventTracker = new RunningDocumentTableEventTracker(workspace._threadingContext,
                     componentModel.GetService<IVsEditorAdaptersFactoryService>(), runningDocumentTable, this);
+
+                _workspaceApplicationQueue = new AsyncBatchingWorkQueue<Func<Task>>(
+                    TimeSpan.FromMilliseconds(250),
+                    ProcessWorkspaceApplicationQueueAsync,
+                    listenerProvider.GetListener(FeatureAttribute.Workspace),
+                    threadingContext.DisposalToken);
+            }
+
+            private async ValueTask ProcessWorkspaceApplicationQueueAsync(ImmutableArray<Func<Task>> taskFactories, CancellationToken cancellationToken)
+            {
+                foreach (var taskFactory in taskFactories)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
```

Comment:
not certain if cancellation is accepatable.  i think it is though as this means VS is shutting down... so we shouldn't continue to do work.  

but i can remove this if it's critical that the workspace (and this type) process all queued requests even on shutdown.

Reply:
thanks!